src/d/actor/d_a_player_hang.inc

```diff
@@ -11,98 +11,613 @@
+                       daPy_HIO_hang_c0::m.field_0x8,
+                       daPy_HIO_hang_c0::m.field_0x0, morf);
+    mVelocity = 0.0f;
+    speed.y = 0.0f;
+    dComIfGp_setPlayerStatus0(0, daPyStts0_UNK100_e);
+    voiceStart(11);
+    return true;
 }
 
 /* 80134DF8-80134EF0       .text procHangFallStart__9daPy_lk_cFv */
 BOOL daPy_lk_c::procHangFallStart() {
-    /* Nonmatching */
+    int iVar2;
+
+    if ((mFrameCtrlUnder[UNDER_MOVE0_e].getRate() < 0.0099999998f) ||
+        (mFrameCtrlUnder[UNDER_MOVE0_e].getFrame() >
+         daPy_HIO_hang_c0::m.field_0xC)) {
+      iVar2 = 1;
+    } else {
+      iVar2 = 0;
+    }
+    if (changeHangEndProc(iVar2)) {
+      return true;
+    } else {
+      if (iVar2 != 0) {
+        if (checkModeFlg(ModeFlg_00000400)) {
+          offModeFlg(ModeFlg_00000400);
+          onModeFlg(ModeFlg_00000100);
+        }
+        if (checkModeFlg(ModeFlg_00000100)) {
+          setTextureAnime(7, 0);
+        }
+        iVar2 = getHangDirectionFromAngle();
+        if ((mStickDistance > 0.05f) && (iVar2 != 1)) {
+          procHangUp_init(iVar2);
+        }
+      }
+    }
+    return true;
 }
 
 /* 80134EF0-80134F74       .text procHangUp_init__9daPy_lk_cFi */
-BOOL daPy_lk_c::procHangUp_init(int) {
-    /* Nonmatching */
+BOOL daPy_lk_c::procHangUp_init(int param_0) {
+    commonProcInit(daPyProc_HANG_UP_e);
+    setSingleMoveAnime(ANM_HANGUP, daPy_HIO_hang_c0::m.field_0x14,
+                       daPy_HIO_hang_c0::m.field_0x18,
+                       daPy_HIO_hang_c0::m.field_0x2,
+                       daPy_HIO_hang_c0::m.field_0x20);
+    dComIfGp_setPlayerStatus0(0, daPyStts0_UNK100_e);
+    m3570 = param_0;
+    mVelocity = 0.0f;
+    return true;
 }
 
 /* 80134F74-80135094       .text procHangUp__9daPy_lk_cFv */
 BOOL daPy_lk_c::procHangUp() {
-    /* Nonmatching */
+    if (changeHangEndProc(1)) {
+      if (mCurProc == daPyProc_FALL_e) {
+        current.pos.x = current.pos.x - 8.0f * cM_ssin(shape_angle.y);
+        current.pos.z = current.pos.z - 8.0f * cM_scos(shape_angle.y);
+      }
+      return true;
+    } else {
+      if ((mFrameCtrlUnder[UNDER_MOVE0_e].getRate() < 0.0099999998f) ||
+          (mFrameCtrlUnder[UNDER_MOVE0_e].getFrame() >
+           daPy_HIO_hang_c0::m.field_0x1C)) {
+        if (m3570 == 0) {
+          procHangClimb_init(daPy_HIO_wallCatch_c0::m.field_0x2C);
+        } else {
+          if (changeHangMoveProc(m3570)) {
+            procHangMove_init(m3570);
+          } else {
+            procHangWait_init();
+          }
+        }
+      }
+    }
+    return true;
 }
 
 /* 80135094-80135140       .text procHangWait_init__9daPy_lk_cFv */
 BOOL daPy_lk_c::procHangWait_init() {
-    /* Nonmatching */
+    commonProcInit(daPyProc_HANG_WAIT_e);
+    //!@bug? passes a signed short (daPy_HIO_wallCatch_c0::m.field_0x2) as a
+    //!float.
+    setSingleMoveAnime(ANM_VJMPCHA, 0.0f, daPy_HIO_wallCatch_c0::m.field_0x2, -1,
```

Comment:
If you pass an integer typed variable as a float argument to a function the compiler should automatically convert it to float type and put it in the right floating point register, this doesn't sound like a bug unless I'm misunderstanding the comment